lib/Transforms/TransformationPassPipelines.cpp

```diff
@@ -35,10 +48,20 @@ void registerTransformationPassPipelines() {
   PassPipelineRegistration<>(
       "llzk-remove-unnecessary-ops-and-defs",
       "Remove unnecessary operations, field definitions, and struct definitions",
-      [](OpPassManager &pm) {
-    pm.addPass(createRedundantReadAndWriteEliminationPass());
-    pm.addPass(createRedundantOperationEliminationPass());
-    pm.addPass(createUnusedDeclarationEliminationPass());
+      [](OpPassManager &pm) { addRemoveUnnecessaryOpsAndDefsPipeline(pm); }
+  );
+
+  mlir::PassPipelineRegistration<FullPolyLoweringOptions>(
+      "llzk-full-poly-lowering",
+      "Lower all polynomial constraints to a given max degree, then remove unnecessary operations "
+      "and definitions.",
+      [](mlir::OpPassManager &pm, const FullPolyLoweringOptions &opts) {
+    // 1. Degree lowering
+    auto polyPass = llzk::createPolyLoweringPass(opts.maxDegree);
+    pm.addPass(std::move(polyPass));
```

Comment:
Does this really need the move or can we do more succinctly as `pm.addPass(llzk::createPolyLoweringPass(opts.maxDegree));`?